portfolio/src/main/java/com/google/sps/servlets/UploadHandlerServlet.java

```diff
@@ -0,0 +1,118 @@
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+import java.util.ArrayList;
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * When the user submits the form, Blobstore processes the file upload and then forwards the request
+ * to this servlet. This servlet can then process the request using the file URL we get from
+ * Blobstore.
+ */
+@WebServlet("/upload-handler")
+public class UploadHandlerServlet extends HttpServlet {
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        ArrayList<String> list = new ArrayList<>();
+        
+        Query query = new Query("Image").addSort("timestamp", SortDirection.DESCENDING);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+        
+        for(Entity entity: results.asIterable()) {
+            String imageUrl = (String) entity.getProperty("poster");
+            list.add(imageUrl);
+        }
+
+        response.setContentType("text/html;");
+        String json = new Gson().toJson(list);
+        response.getWriter().println(json);
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        // Get the URL of the image that the user uploaded to Blobstore.
+        String imageUrl = getUploadedFileUrl(request, "movie-poster");
+
+        System.out.println("image url: " + imageUrl);
+        
+        Entity posterEntity = new Entity("Image");
+        posterEntity.setProperty("poster", imageUrl);
+        long timestamp = System.currentTimeMillis();
+        posterEntity.setProperty("timestamp", timestamp);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        datastore.put(posterEntity);
+
+        response.sendRedirect("/index.html");
+
+         // Output some HTML that shows the data the user entered.
+
+        PrintWriter out = response.getWriter();
+        out.println("<p>Here's the image you uploaded:</p>");
+        out.println("<a href=\"" + imageUrl + "\">");
+        out.println("<img src=\"" + imageUrl + "\" />");
+        out.println("</a>");
```

Comment:
It would probably be better in this case to create the HTML element in the frontend and only return the imageURL